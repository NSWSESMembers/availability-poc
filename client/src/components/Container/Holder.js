import PropTypes from 'prop-types';
import React from 'react';
import { View } from 'react-native';
import styles from './styles';

const Container = ({
  children,
  margin,
  marginVertical,
  marginTop,
  paddingVertical,
  wide,
  marginBot,
  transparent,
}) => {
  const containerStyles = [styles.containerHolder];

  if (paddingVertical) {
    containerStyles.push({ paddingTop: 8, paddingBottom: 8 });
  }

  if (marginVertical) {
    containerStyles.push({ marginTop: 8, marginBottom: 8 });
  }

  if (marginTop) {
    containerStyles.push({ marginTop: 8 });
  }

  if (margin) {
    containerStyles.push({ paddingTop: 20 });
  }

  if (wide) {
    containerStyles.push({ paddingLeft: 5, paddingRight: 5 });
  }

  if (marginBot) {
    containerStyles.push({ paddingBottom: 20 });
  }
  if (transparent) {
    containerStyles.push({ backgroundColor: 'transparent' });
  }

  return <View style={containerStyles}>{children}</View>;
};

Container.propTypes = {
  children: PropTypes.node.isRequired,
  paddingVertical: PropTypes.bool,
  marginTop: PropTypes.bool,
  marginVertical: PropTypes.bool,
  margin: PropTypes.bool,
  marginBot: PropTypes.bool,
  wide: PropTypes.bool,
  transparent: PropTypes.bool,
};

export default Container;
