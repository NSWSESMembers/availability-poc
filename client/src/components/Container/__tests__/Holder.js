import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import Holder from '../Holder';

test('renders correctly', () => {
  const tree1 = renderer.create(
    <Holder>
      <Text>test</Text>
    </Holder>,
  ).toJSON();
  expect(tree1).toMatchSnapshot();

  const tree2 = renderer.create(
    <Holder
      paddingVertical
      marginVertical
      marginTop
      margin
      wide
      marginBot
      transparent
    >
      <Text>test</Text>
    </Holder>,
  ).toJSON();
  expect(tree2).toMatchSnapshot();
});
