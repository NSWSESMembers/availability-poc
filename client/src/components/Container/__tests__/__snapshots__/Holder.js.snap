// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "paddingLeft": 20,
        "paddingRight": 20,
        "width": "100%",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    test
  </Text>
</View>
`;

exports[`renders correctly 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "paddingLeft": 20,
        "paddingRight": 20,
        "width": "100%",
      },
      Object {
        "paddingBottom": 8,
        "paddingTop": 8,
      },
      Object {
        "marginBottom": 8,
        "marginTop": 8,
      },
      Object {
        "marginTop": 8,
      },
      Object {
        "paddingTop": 20,
      },
      Object {
        "paddingLeft": 5,
        "paddingRight": 5,
      },
      Object {
        "paddingBottom": 20,
      },
      Object {
        "backgroundColor": "transparent",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    test
  </Text>
</View>
`;
